# This is a patch to disable the rule "Contact K8S API Server From Container" in the falco_rules.yaml file.
# Otherwise, we get: "LOAD_ERR_COMPILE_CONDITION (Error compiling condition): filter_check called with nonexistent field k8s.pod.name" error.
# TODO(Dentrax): Ask to maintainers to fix this issue.
diff --git a/rules/falco_rules.yaml b/rules/falco_rules.yaml
index 12ee70d..af10bce 100644
--- a/rules/falco_rules.yaml
+++ b/rules/falco_rules.yaml
@@ -820,25 +820,6 @@
 - macro: user_known_contact_k8s_api_server_activities
   condition: (never_true)
 
-- rule: Contact K8S API Server From Container
-  desc: >
-    Detect attempts to communicate with the K8S API Server from a container by non-profiled users. Kubernetes APIs play a 
-    pivotal role in configuring the cluster management lifecycle. Detecting potential unauthorized access to the API server 
-    is of utmost importance. Audit your complete infrastructure and pinpoint any potential machines from which the API server 
-    might be accessible based on your network layout. If Falco can't operate on all these machines, consider analyzing the 
-    Kubernetes audit logs (typically drained from control nodes, and Falco offers a k8saudit plugin) as an additional data 
-    source for detections within the control plane.
-  condition: >
-    evt.type=connect and evt.dir=< 
-    and (fd.typechar=4 or fd.typechar=6) 
-    and container 
-    and k8s_api_server 
-    and not k8s_containers 
-    and not user_known_contact_k8s_api_server_activities
-  output: Unexpected connection to K8s API Server from container (connection=%fd.name lport=%fd.lport rport=%fd.rport fd_type=%fd.type fd_proto=fd.l4proto evt_type=%evt.type user=%user.name user_uid=%user.uid user_loginuid=%user.loginuid process=%proc.name proc_exepath=%proc.exepath parent=%proc.pname command=%proc.cmdline terminal=%proc.tty exe_flags=%evt.arg.flags %container.info)
-  priority: NOTICE
-  tags: [maturity_stable, container, network, k8s, mitre_discovery, T1565]
-
 - rule: Netcat Remote Code Execution in Container
   desc: > 
     Netcat Program runs inside container that allows remote code execution and may be utilized 
